### PR TITLE
SI-6388 Removes deprecated functionality for 2.11

### PR DESCRIPTION
**Pull request by soc**
Not in this commit:
- Deprecated methods of List (does Paul still want to keep them?)
- @serializable annotation (removal needs multiple steps)

### CODE REVIEW

File: src/compiler/scala/tools/nsc/Global.scala [hunk 1]
```diff
@@ -1733,20 +1733,12 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
...
-  // and forScaladoc default to onlyPresentation, which is the same as defaulting
-  // to false except in old code.  The downside is that this leaves us calling a
-  // deprecated method: but I see no simple way out, so I leave it for now.
+
   def forJVM           = settings.target.value startsWith "jvm"
   override def forMSIL = settings.target.value startsWith "msil"
-  def forInteractive   = onlyPresentation
-  def forScaladoc      = onlyPresentation
+  def forInteractive   = false
+  def forScaladoc      = false
```
Comment (soc) [inline src/compiler/scala/tools/nsc/Global.scala:1747]: Is this the right solution?

**Comment by soc** [comment]
I just tested it without the change in Entity.scala:L153 and it still fails ...

**Comment by soc** [comment]
The cause is in the first commit, but it is not `inheritedFrom` ... weird.

**Comment by paulp** [comment]
Some of these need to be kept longer.  At least: src/compiler/scala/tools/nsc/Interpreter.scala,InterpreterLoop.scala.  Have only started looking at it.

**Comment by paulp** [comment]
Okay, I think that's it.  The List methods can go.  The rest LGTM but a second pair of eyes would be good.

File: src/compiler/scala/tools/nsc/Global.scala (hunk 1 above)
Comment (gkossakowski) [inline src/compiler/scala/tools/nsc/Global.scala:1747]: @VladUreche: can you comment on this fragment?

**Comment by gkossakowski** [comment]
I'd prefer if we had more fine-grained commits. The bare minimum is to split changes made to compiler and library. The best would be that you go one package at the time. The reason is that I want to be able to easily pinpoint the exact change that broke some tool and revert it if needed without reverting everything else.

For this kind of changes I totally don't mind having 5 or 10 focused commits. I actually prefer that.

File: src/compiler/scala/tools/nsc/Global.scala (hunk 1 above)
Comment (VladUreche) [inline src/compiler/scala/tools/nsc/Global.scala:1747]: This change LGTM, it's overridden in DocFactory anyway.

**Comment by soc** [comment]
@gkossakowski You're absolutely right. I'm currently splitting up the commits into smaller parts and test them individually.

As a first benefit of having smaller commits, I was able to reduce the amount of suspects involved in the ScalaDoc test failure. It really looks like as if either the removal of `Application` or the items in `Predef` are causing the error.

I still have no idea how this happens, though ...

**Pull request closed without merging**